python/amici/petab_objective.py

```diff
@@ -159,10 +178,116 @@ def simulate_petab(
 
     return {
         LLH: llh,
+        SLLH: sllh,
         RDATAS: rdatas
     }
 
 
+def aggregate_sllh(
+        amici_model: AmiciModel,
+        rdatas: Sequence[amici.ReturnDataView],
+        parameter_mapping: Optional[ParameterMapping],
+        edatas: List[AmiciExpData],
+        petab_scale: bool = True,
+        petab_problem: petab.Problem = None,
+) -> Union[None, Dict[str, float]]:
+    """
+    Aggregate likelihood gradient for all conditions, according to PEtab
+    parameter mapping.
+
+    :param amici_model:
+        AMICI model from which ``rdatas`` were obtained.
+    :param rdatas:
+        Simulation results.
+    :param parameter_mapping:
+        PEtab parameter mapping to condition-specific
+            simulation parameters
+    :param edatas:
+        Experimental data used for simulation.
+    :param petab_scale:
+        Return sensitivities on the PEtab parameter scales
+    :param petab_problem:
+        The PEtab problem that defines the parameter scales.
+
+    :return:
+        Aggregated likelihood sensitivities.
+    """
+    accumulated_sllh = {}
+    model_parameter_ids = amici_model.getParameterIds()
+
+    if petab_scale and petab_problem is None:
+        raise ValueError(
+            'Please provide a PEtab problem if scaled SLLH is desired, or set '
+            '`petab_scale==False`.'
+        )
+
+    # Check for issues in all condition simulation results.
+    for rdata in rdatas:
+        # Condition failed during simulation.
+        if rdata['status'] != amici.AMICI_SUCCESS:
+            return None
+        # Condition simulation result does not provide SLLH.
+        if rdata.get('sllh', None) is None:
+            return None
```

Comment:
one could argue about whether to silently return None or cry out if sth is inconsistent, but anything fine for me.